src/components/partnership/partnership.service.ts

```diff
@@ -326,38 +321,10 @@ export class PartnershipService {
 
       const partnership = await this.readOne(result.id, session);
 
-      if (!input.types) {
-        return partnership;
-      }
```

Comment:
This was (incorrectly) done to handle types being null, when they are defined as an array. Can you handle this? Maybe the DTO gets a default value or maybe a check is done on the DB result to ensure it's an array?

Reply:
How's about this for handling an empty array.
```
if (!input.types || isEmpty(input.types)) {
   return partnership;
}
```

Reply:
Sorry for the confusion. The early return is wrong. We always want to publish the event. 

It's the fact that we have `types` as an array but it was sometimes `null`. The change you made on the DTO might be enough. If it's not, then we should ensure that it's an array here. But we should always publish the event.

Reply:
Okay got it

Reply:
I think changing on DTO is enough because `readOne` will return `[]` if the `types` is `null`.